lib/services/file/fileservice.core.js

```diff
@@ -450,6 +506,8 @@ FileService.prototype.createShareIfNotExists = function (share, optionsOrCallbac
   });
 
   var options = extend(true, {}, userOptions);
+  delete options.shareSnapshotId;
```

Comment:
Why delete, but not validate during validateArgs

Reply:
We delete it, because we need pass the options into _doesFileExist.